distributed/deploy/spec.py

```diff
@@ -383,7 +383,10 @@ async def _close(self):
             await future
         async with self._lock:
             with ignoring(CommClosedError):
-                await self.scheduler_comm.close(close_workers=True)
+                if self.scheduler_comm:
+                    await self.scheduler_comm.close(close_workers=True)
+                else:
+                    logger.warning("Cluster closed without starting up")
```

Comment:
Are there other (typical) situations where we might end up with no `.scheduler_comm` attribute? If not then this seems good.